hack/platform-validation.sh

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+#
+# This file is part of the Kepler project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Copyright 2023 The Kepler Contributors
+#
+
+set -e
+export CLUSTER_PROVIDER=${CLUSTER_PROVIDER:-kubernetes}
+# set the CLUSTER_PROVIDER to kubernetes or openshift kind
+source ./hack/common.sh
+
+export IMAGE_TAG=${IMAGE_TAG:-devel}
+export IMAGE_REPO=${IMAGE_REPO:-quay.io/sustainable_computing_io}
+
+# we do not use `make cluster-clean` and `make cluster-deploy` because they trigger other actions
+function main() {
+    make build-manifest OPTS="PROMETHEUS_DEPLOY"
+
+    ./hack/cluster-clean.sh &
+    CLEAN_PID=$!
+
+    make build_containerized
+    make push-image
+
+    echo "waiting for cluster-clean to finish"
+    if ! wait $CLEAN_PID; then
+        echo "cluster-clean failed"
+    fi
+
+    ./hack/cluster-deploy.sh
+
+    #TODO: run validation cases and generate report.
```

Comment:
where are we plan to store the report?

Reply:
Good question, originally I am intent to add the test cases code in shell and generate report there.
But after reviewing current kepler integration test code, maybe better idea is to add a new action like .github/workflow/platform-validation.yaml, and write specific go test code under kepler/e2e/ directory?
The report is another open topic for community, store under kepler/test directory? and how about the report content format? We are thinking about this and may submit proposal for community review in the following weeks..